pybossa/sched.py

```diff
@@ -376,7 +403,7 @@ def get_locked_task(project_id, user_id=None, limit=1, rand_within_priority=Fals
 
 @locked_scheduler
 def get_user_pref_task(project_id, user_id=None, limit=1, rand_within_priority=False,
-                       task_type='gold_last'):
+                       task_type='gold_last', filter_user_prefs=True):
```

Comment:
Why are we defaulting filter_user_prefs to True?

Reply:
because this function is used only when the scheduler is "User Preference Scheduler", it should always filter by user preference

Reply:
It is passed in `@locked_scheduler` function